### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by benediamond**
I am getting better results after switching from `SGD` to the `Adam` optimizer. After about 12 hours of supervised training, I have losses of around
`loss: 3.4352 - policy_out_loss: 2.1883 - value_out_loss: 0.2991`
and they appear to be continuing to improve. I am using 7 residual layers, 4672 output labels, and 21x8x8 input (AlphaZero style with no history, T = 1). But GUI play was not great.

I also fixed an important bug in the MCTS, see [this issue](https://github.com/mokemokechicken/reversi-alpha-zero/issues/17).

### CODE REVIEW

**Comment by Zeta36** [comment]
Great, @benediamond!! I'm really glad to hear that :).  How are you training the model? Using supervised learning? If the GUI play is not good maybe you are overfitting. How many PGN files are you using?

**Comment by benediamond** [comment]
I am using a KingBase file with ~7000 games. But I only keep 20 .json files in memory at once, each with 100 games. This is so that memory consumption doesn't get too large. But I cycle files in and out on a rolling basis. Loss climbed a bit while files were rolling in and out, but not by too much.

**Comment by Zeta36** [comment]
You are overfitting. You need larger files. I use 4Gb of human games.

**Comment by Zeta36** [comment]
Even if you use only 20 files at a time, you need much more games than 7000. That's problably the cause of your poor GUI game.

**Comment by Akababa** [comment]
Good catch on the q part! I think overfitting also has to do with the number of epochs, I trained last night on 350,000 positions for 5 epochs and the result was still overfitted.

**Comment by benediamond** [comment]
As far as overfitting, @Akababa you're right that actually training this would require massive computing power. The mere fact that my loss dropped, even if it might be overfitted, shows me that at least the network should be capable of learning "in principle" and it may be free from major bugs. I hope. Incidentally, 20 files with 100 games each is just about exactly 4GB, a little more.

**Comment by Akababa** [comment]
One thing that may help is training on games where the elo difference between the players is large, and ignoring the policy of the weaker player. I suspect this might help the model learn the value of winning a queen..

**Comment by Zeta36** [comment]
DeepMind for the first version of AlphaGo used of millions of humans complete games for training the policy network so I guess just ~10.000 are far away of a decent result.

About the 4GB I was referring to the PGN files size. In 4GB of raw PGN you have hundreds of thousands of games for shuffling.

**Comment by Akababa** [comment]
That's true, AZ used a queue of the last 500k self-play games for training. You can also drop the memory usage by using float32 and losing the filenames dict.

BTW when do you think we'll be ready to start a distributed effort? Should we have a validation/test set?
